### PR TITLE
Copy Repl project into PowerFx repo 

### PR DESCRIPTION
**Pull request by MikeStall**
From Samples:
https://github.com/microsoft/power-fx-host-samples/blob/main/Samples/ConsoleREPL/ConsoleREPL.cs

We still want Samples copy since that goes against actual nugets. 
But a copy in the PowerFx repo is convenient for directly testing the SDK. 

Lots of whitespace differences because I had to bring up to coding conventions for this repo. 

Also leverage some recent functionality:
- replace a hack Set() RegEx with using official APIs (parse, etc) 
- FormulaValue.ToExpression to get  string from arbitrary expression.

### CODE REVIEW

File: src/tools/Repl/Program.cs [hunk 1]
```diff
@@ -0,0 +1,601 @@
...
+
+            Console.WriteLine($"Experimental features enabled:{enabled}");
+
+            Console.WriteLine($"Enter Excel formulas.  Use \"Help()\" for details.");
+
+            REPL(Console.In, false);
+        }
+
+        // Pattern match for Set(x,y) so that we can define the variable
+        public static bool TryMatchSet(string expr, out string arg0name, out FormulaValue varValue)
```
Comment (MikeStall) [inline src/tools/Repl/Program.cs:80]: This removes a hacky Set regex and uses "Official" APIs. 


Also see
https://github.com/microsoft/Power-Fx/discussions/889

File: src/tools/Repl/Program.cs [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+                try
+                {
+                    // variable assignment: Set( <ident>, <expr> )
+                    if (TryMatchSet(expr, out var varName, out var varValue))
+                    {
+                        Console.WriteLine(varName + ": " + PrintResult(varValue));
+                    }
+
+                    // formula definition: <ident> = <formula>
+                    else if ((match = Regex.Match(expr, @"^\s*(?<ident>\w+)\s*=(?<formula>.*)$", RegexOptions.Singleline)).Success)
```
Comment (MikeStall) [inline src/tools/Repl/Program.cs:141]: Would like to get rid of these RegEx, but the corresponding Parse functions are still internal...

File: src/tools/Repl/Program.cs [hunk 3]
```diff
@@ -0,0 +1,601 @@
...
+                            separator = ", ";
+                        }
+
+                        resultString += "]";
+                    }
+                }
+            }
+            else
+            {
+                resultString = value.ToExpression();
```
Comment (MikeStall) [inline src/tools/Repl/Program.cs:433]: This is also a nice simplification - we now have more reliable ToString functions for FormulaValues.

**Review by LucGenetier — DISMISSED** [review]

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by LucGenetier — APPROVED** [review]

**Pull request merged**